pkg/util/cost_attribution.go

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package util
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/services"
+	"go.uber.org/atomic"
+)
+
+type CostAttribution struct {
```

Comment:
This looks very similar to `pkg/util/active_groups.go`, as mentioned offline: both should follow the same logic and use the same label.